### DISCUSSION TITLE
PS7 MSI installation - How to detect USE_MU=1?

### DISCUSSION THREAD

**Discussion by CharlesNRU** [PowerShell/PowerShell#19271] [Q&A]
Hello,

I would like to deploy PowerShell 7 to multiple users at my company via SCCM.
As part of the installation process, I need to be able to properly "detect" that PS7 was installed successfully. For a MSI like PS7, it is usually easy to use as a detection method. However, I also want to enable updating PowerShell 7 through Microsoft Update, WSUS or SCCM. As indicated in the installation documentation, I can use the USE_MU=1 property when installing PowerShell (https://learn.microsoft.com/en-us/powershell/scripting/install/installing-powershell-on-windows?view=powershell-7.3#support-for-microsoft-update-in-powershell-72-and-newer).

As part of my detection script, I would like to confirm that this was also enabled but I cannot find a way to "detect" that PS7 will be updated via updates. How can I confirm that whatever USE_MU=1 does actually worked? I searched in the registry to find a key or something that would indicate that this feature is enabled but I couldn't find anything.

Thanks.

**Comment by iSazonov** [comment]
It is regular update like any another Windows update. So you can use Windows Update Agent (WUA) API to manually check status of updates. There is PowerShell module(s) to access the API.

If you distribute updates by WSUS or SCCM they have internal reports for every update per computer.

**Reply by CharlesNRU** [reply]
I was actually looking for a way to detect if USE_MU was enabled or not on an existing PS7 installation.
I was able to find the answer after some digging.
There's a registry key at HKLM:\Software\Microsoft\PowerShellCore named UseMU set to 1.
One thing I did find is that the recommended command to fix this on an existing installation doesn't work.
`msiexec.exe /fmu .\PowerShell-7.3.0-win-x64.msi USE_MU=1 ENABLE_MU=1`
Source: https://learn.microsoft.com/en-us/powershell/scripting/install/microsoft-update-faq?view=powershell-7.3#i-haven-t-received-an-update-for-the-new-release--why-not

When I run this command, it does not change the registry key.

**Reply by iSazonov** [reply]
Please open new issue with repro steps.
